databuilder/main.py

```diff
@@ -169,9 +175,30 @@ def add_to_sys_path(directory):
         sys.path = original
 
 
-def write_dataset_csv(column_specs, results, dataset_file):
+def get_file_extension(filename):
+    if filename is None:
+        # If we have no filename we're writing to stdout, so default to CSV
+        return ".csv"
+    elif filename.suffix == ".gz":
+        return "".join(filename.suffixes[-2:])
+    else:
+        return filename.suffix
+
+
+def write_dataset_csv(column_specs, results, dataset_file, gzipped=False):
     headers = list(column_specs.keys())
-    with open_output_file(dataset_file) as f:
+    # Set `newline` as per Python docs: https://docs.python.org/3/library/csv.html#id3
+    with open_output_file(dataset_file, newline="", gzipped=gzipped) as f:
```

Comment:
thought: I was going to ask about output CSV line endings, and don't need to know now :+1: